test/new-e2e/tests/otel/utils/config_utils.go

```diff
@@ -70,3 +84,24 @@ func getAgentPod(s OTelTestSuite) corev1.Pod {
 	require.NotEmpty(s.T(), res.Items)
 	return res.Items[0]
 }
+
+func validateConfigs(t *testing.T, expectedCfg string, actualCfg string) {
+	var actualConfRaw map[string]any
+	require.NoError(t, yaml.Unmarshal([]byte(actualCfg), &actualConfRaw))
+
+	// traces, metrics and logs endpoints are set to the fake intake address in the config
+	// these endpoints should be ignored in the comparison
+	exps, _ := actualConfRaw["exporters"].(map[string]any)
+	ddExp, _ := exps["datadog"].(map[string]any)
+	tcfg := ddExp["traces"].(map[string]any)
+	delete(tcfg, "endpoint")
+	mcfg := ddExp["metrics"].(map[string]any)
+	delete(mcfg, "endpoint")
+	lcfg := ddExp["logs"].(map[string]any)
+	delete(lcfg, "endpoint")
+
+	var expectedConfRaw map[string]any
+	require.NoError(t, yaml.Unmarshal([]byte(expectedCfg), &expectedConfRaw))
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
There is a [assert.YAMLEq](https://pkg.go.dev/github.com/stretchr/testify/assert#YAMLEq) you might use instead.

Still, this would fail if orders in the map values are different, as in

```yaml
tags:
	- a
	- b
```

vs

```yaml
tags:
	- b
	- a
```

See this [go playground](https://goplay.tools/snippet/twVxOMEgEHB) for an example 

@KevinFairise2 recently cleaned up a test I broke in test-infra, using https://github.com/DataDog/test-infra-definitions/blob/main/common/utils/yaml_test.go#L91-L103. We might move the function to some helper that could be used here. In the meantime you could copy paste it, if it makes sense

Reply:
`assert.YAMLEq` works, used it instead